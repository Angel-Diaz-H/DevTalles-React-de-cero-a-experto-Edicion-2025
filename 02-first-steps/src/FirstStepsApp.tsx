import { ItemCounter } from './shopping-cart/itemCounter';

export function FirstStepsApp() {
    return (
        <>
            {/* <h1>Hola!!!</h1>
        <p>Esto es un p√°rrafo.</p>
        <button>Click me</button> */}

            <h1>Carrito de compras</h1>
            <ItemCounter name='Nintendo Switch' quantity={3} />
            <ItemCounter name='Xbox' quantity={2} />
        </>
    )
};