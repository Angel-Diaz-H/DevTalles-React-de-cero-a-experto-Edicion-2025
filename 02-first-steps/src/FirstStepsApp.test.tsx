import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";

describe('FirstSteppApp', () => {
    test('Should match snapshot', () => {
        const { container } = render(<FirstStepsApp />)
        expect(container).toMatchSnapshot();

        screen.debug();
    });
});