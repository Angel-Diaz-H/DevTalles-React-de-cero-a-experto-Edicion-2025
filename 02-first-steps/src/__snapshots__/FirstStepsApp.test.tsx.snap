// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstSteppApp > Should match snapshot 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Nintendo Switch
    </span>
    <button>
      +1
    </button>
    <span>
      4
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Pro controller
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Xbox
    </span>
    <button>
      +1
    </button>
    <span>
      5
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
